source/dofs/dof_tools_constraints.cc

```diff
@@ -546,91 +546,56 @@ namespace DoFTools
     } // namespace
 
 
-    template <typename number>
-    void
-    make_hp_hanging_node_constraints(const dealii::DoFHandler<1> &,
-                                     AffineConstraints<number> &)
-    {
-      // nothing to do for regular dof handlers in 1d
-    }
 
-
-    template <typename number>
-    void
-    make_oldstyle_hanging_node_constraints(const dealii::DoFHandler<1> &,
-                                           AffineConstraints<number> &,
-                                           std::integral_constant<int, 1>)
-    {
-      // nothing to do for regular dof handlers in 1d
-    }
-
-
-    template <typename number>
+    template <typename number, int spacedim>
     void
     make_hp_hanging_node_constraints(
-      const dealii::hp::DoFHandler<1> & /*dof_handler*/,
+      const dealii::hp::DoFHandler<1, spacedim> & /*dof_handler*/,
       AffineConstraints<number> & /*constraints*/)
     {
       // we may have to compute constraints for vertices. gotta think about that
       // a bit more
 
       // TODO[WB]: think about what to do here...
+      Assert(false, ExcNotImplemented());
```

Comment:
Why did you add these?

Reply:
I added these, because it is not obvious to me that there are generally no constraints for the hp case in 1d stemming from p-adaptivity (see also the comment in the code of @bangerth). If this is really not the case, the assertion can be eliminated (and also some template specializations joined because then there is no difference between DoFHandler and hp::DoFHandler for 1d anymore).

Reply:
That comment is ~13 years old. I haven't given it any dedicated thought, but I believe that I would have come across a case where we need these constraints since then. So I think both the comment and the assertion can go.

Reply:
All the better, then these assertions are superfluous and the PR should also not break tests anymore :)